src/wp-includes/class-wp-scripts.php

```diff
@@ -648,7 +681,18 @@ private function should_delay_inline_script( $handle, $position ) {
 		 * dependency's after inline script.
 		 */
 		foreach ( $deps as $dep ) {
-			if ( $this->is_delayed_strategy( $this->get_eligible_loading_strategy( $dep ) ) ) {
+			if ( ! array_key_exists( $dep, $this->registered ) ) {
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/4391/files?file-filters%5B%5D=.php&show-viewed-files=true#r1232052943

Reply:
Done in 47b2cbccfc7641d141994c5ec0a28e356d12da3e